### PR TITLE
Bob/5817 return focus on archive close

### LINKED ISSUE #5817: Archive patient modal does not return focus to when exiting modal
## Area or Component

[Patients page - Archive patient modal](https://demo.simplereport.gov/app/patients/1?facility=14fab3c1-070e-4e86-8ec7-4a1fc8e13477)
## Quick Summary

When exiting out of the modal via cancel or escape key, focus should be returned to the archive patient button instead of the Patient search box.

### How to Replicate

Click the three dots button next to patient, click the archive patient, and then either click "No, go back" or escape button.

https://github.com/CDCgov/prime-simplereport/assets/10108172/9afe7fc3-e5e7-422d-a50e-1fb8cceb894d

## Accessibility Violation

[Focus modal not returned](https://docs.deque.com/issue-help/1.0.0/en/focus-modal-not-returned)

## Action Requested

On cancel or escape of the modal, focus should be returned back to

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue
Fixes #5817 

## Changes Proposed

- Returns focus to the action menu of the relevant user when the archive modal is closed
- Adds a corresponding test
- Retain the default focus to the patient search bar 
- Decided to refocus on the action menu for the user row in question rather than the explicit archive button, since 
   - I'm guessing that when users go back from the archive menu, moving back to the action menu would allow them to redirect themselves as needed (ie tab to the next user row, move to the "Start test" option instead) better than the actual archive button in the expanded list.
   - The secondary action dropdown isn't expanded on modal exit and it didn't seem worth it to expand it programmatically in this case

## Testing

- deployed on dev 2
- To replicate
    - Navigate to the manage patients menu
    - Select archive a patient
    - Exit out of the modal and notice focus returns to the action button

## Screenshots / Demos
https://github.com/CDCgov/prime-simplereport/assets/29645040/75d507f4-250f-437d-b9d3-021153013123

### Accessibility
- [x] Any large changes have been run through Deque manual testing
- [x] All changes have run through the Deque automated testing

### CODE REVIEW

**Review by BobanL — APPROVED** [review]
Changes look good and passes accessibility!

File: frontend/src/app/patients/ManagePatients.test.tsx [hunk 1]
```diff
@@ -107,6 +107,19 @@ describe("ManagePatients", () => {
       ).toBeInTheDocument();
     });
 
+    it("when exiting archive modal, the action button is refocused", async () => {
+      render(<TestContainer />);
+      expect(await screen.findByText(patients[0].lastName, { exact: false }));
+
+      const menu = (await screen.findAllByText("More actions"))[0];
```
Comment (johanna-skylight) [inline frontend/src/app/patients/ManagePatients.test.tsx:114]: Can you wrap these actions in the act method or switch them to fireEvent.click? It is to not get the act warnings . You can add an `.only` to the it method to only run the new added coverage `it.only(` so you can test it isolated!

Comment (fzhao99) [reply frontend/src/app/patients/ManagePatients.test.tsx:114]: So I did a little bit of digging around this because it was annoying me and I think I found that the root cause is a package versioning mismatch. Expo [here](https://github.com/testing-library/user-event/issues/1104#issuecomment-1460005394). 

TL;DR is that our versions of `@testing-library/dom` need to match throughout the tree in order to not generate unnecessary act warnings.

Right now, running the yarn list on  `@testing-library/dom` gives us the following:

```
├─ @testing-library/dom@8.19.0
├─ @testing-library/react@14.0.0
│  └─ @testing-library/dom@9.3.0
└─ react-select-event@5.5.1
   └─ @testing-library/dom@9.3.0
```
Once I updated the root installation of `@testing-library/dom` to 9.3.0, the errors went away, even when using the `userEvent` action. For now, I just went ahead and replaced the one troublesome assertion with `fireEvent` so as to not add new errors, but hopefully the other errors will go away once we upgrade.

Comment (johanna-skylight) [reply frontend/src/app/patients/ManagePatients.test.tsx:114]: ooo this is cool so if we upgrade @testing-library/dom to 9.3.0 the errors will get fixed in the other files as well?. I wonder why we haven't receive this upgrade from dependabot 🤔

Comment (fzhao99) [reply frontend/src/app/patients/ManagePatients.test.tsx:114]: Yeah, when I updated locally the other errors in the file also went away. I'm guessing that it's in the queue per my question [here](https://skylight-hq.slack.com/archives/C01LTSNKEPP/p1686063839120479).

Comment (fzhao99) [reply frontend/src/app/patients/ManagePatients.test.tsx:114]: Screenshot 2023-06-06 at 10 42 17 AM

looks like 9.3.0 was caught in yesterday's scan, so i'm guessing that i'll show up in the next round of updates

**Review by johanna-skylight — APPROVED** [review]
LGTM!

**Review by BobanL — APPROVED** [review]
👍
